packages/core/src/tools/schema.js

```diff
@@ -54,21 +55,26 @@ const convertResourceDos = (appRaw) => {
     }
 
     if (search && create && isVisible(search) && isVisible(create)) {
-      const searchOrCreate = {
-        // key: `${resource.key}SearchOrCreate`,
+      // searchAndCreates is an alias for searchOrCreates. Schema validation makes sure only one of them is defined.
+      const searchCreates = appRaw.searchAndCreates
+        ? appRaw.searchAndCreates
+        : appRaw.searchOrCreates;
+
+      searchCreates[search.key] = {
         key: `${search.key}`, // For now this is a Zapier editor limitation (has to match search)
         display: {
-          label: `Find or Create ${resource.noun}`,
+          label: appRaw.searchAndCreates
+            ? `Upsert ${resource.noun}`
+            : `Find or Create ${resource.noun}`,
           description: _.get(search, ['display', 'description'], ''),
         },
         search: search.key,
         create: create.key,
       };
-      searchOrCreates[searchOrCreate.key] = searchOrCreate;
     }
   });
 
-  return { triggers, searches, creates, searchOrCreates };
+  return { triggers, searches, creates, searchOrCreates, searchAndCreates };
```

Comment:
We may only want to add `searchOrCreates` _or_ `searchAndCreates`, not both. The backend can handle when both are defined, but I'd like it to enforce schema on write.